pkg/providers/bedrock/chat.go

```diff
@@ -0,0 +1,140 @@
+package bedrock
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	//"glide/pkg/providers/clients"
+
+	"glide/pkg/api/schemas"
+
+	"go.uber.org/zap"
+
+	"github.com/google/uuid"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/bedrockruntime"
+)
+
+type ChatMessage struct {
+	Role    string `json:"role"`
+	Content string `json:"content"`
+}
+
+// ChatRequest is an Bedrock-specific request schema
+type ChatRequest struct {
+	Messages             string               `json:"inputText"`
+	TextGenerationConfig TextGenerationConfig `json:"textGenerationConfig"`
+}
+
+type TextGenerationConfig struct {
+	Temperature   float64  `json:"temperature"`
+	TopP          float64  `json:"topP"`
+	MaxTokenCount int      `json:"maxTokenCount"`
+	StopSequences []string `json:"stopSequences,omitempty"`
+}
+
+// NewChatRequestFromConfig fills the struct from the config. Not using reflection because of performance penalty it gives
+func NewChatRequestFromConfig(cfg *Config) *ChatRequest {
+	return &ChatRequest{
+		TextGenerationConfig: TextGenerationConfig{
+			MaxTokenCount: cfg.DefaultParams.MaxTokens,
+			StopSequences: cfg.DefaultParams.StopSequence,
+			Temperature:   cfg.DefaultParams.Temperature,
+			TopP:          cfg.DefaultParams.TopP,
+		},
+	}
+}
+
+func NewChatMessagesFromUnifiedRequest(request *schemas.UnifiedChatRequest) string {
+	// message history not yet supported for AWS models
+	message := fmt.Sprintf("Role: %s, Content: %s", request.Message.Role, request.Message.Content)
+
+	return message
+}
+
+// Chat sends a chat request to the specified bedrock model.
+func (c *Client) Chat(ctx context.Context, request *schemas.UnifiedChatRequest) (*schemas.UnifiedChatResponse, error) {
+	// Create a new chat request
+	chatRequest := c.createChatRequestSchema(request)
+
+	chatResponse, err := c.doChatRequest(ctx, chatRequest)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(chatResponse.ModelResponse.Message.Content) == 0 {
+		return nil, ErrEmptyResponse
+	}
+
+	return chatResponse, nil
+}
+
+func (c *Client) createChatRequestSchema(request *schemas.UnifiedChatRequest) *ChatRequest {
+	// TODO: consider using objectpool to optimize memory allocation
+	chatRequest := c.chatRequestTemplate // hoping to get a copy of the template
+	chatRequest.Messages = NewChatMessagesFromUnifiedRequest(request)
+
+	return chatRequest
+}
+
+func (c *Client) doChatRequest(ctx context.Context, payload *ChatRequest) (*schemas.UnifiedChatResponse, error) {
+	rawPayload, err := json.Marshal(payload)
+	if err != nil {
+		return nil, fmt.Errorf("unable to marshal chat request payload: %w", err)
+	}
+
+	cfg, _ := config.LoadDefaultConfig(ctx,
+		config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(c.config.AccessKey, c.config.SecretKey, "")),
```

Comment:
AWS is full of magic ways to authenticate 😌  I just want to make sure this override won't prevent it from using other authentication approaches like one based on ARN annotation added on the k8s service account resource level (https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html). This is likely to be a production way to give Glide access to Titan.

Reply:
Should I implement this?

Reply:
@mkrueger12 normally AWS SDK comes with support for all of these. But I would double check this override doen't break these other authentication approaches. 

Because the problem with these Access/Secret keys is that it gets expired in 6-12h, so it cannot be really used for setups other than local testing.

Reply:
Do access keys expire that fast if they are attached to an IAM role? I am showing keys that are still active after 3 days.

Reply:
@roma-glushko So I just tested this and the Access & secret keys are still live. We can go a different auth route but this way does work. Let me know what you think.

Reply:
That's good to know. Maybe that's how people operate if they want to use Bedrock while their infra is outside of AWS (e.g. Azure or GCP). 

I will try to inspect a pod that has ARN configured to its service account to see what AWS mounts there additionally to make sure you are with that usecase.